src/main/java/com/library/service/impl/RabbitProducer.java

```diff
@@ -0,0 +1,71 @@
+package com.library.service.impl;
+
+import com.library.config.RabbitProperties;
+import com.library.model.dto.ResponseException;
+import com.library.model.dto.ResponseMessage;
+import lombok.RequiredArgsConstructor;
+import org.aspectj.lang.JoinPoint;
+import org.aspectj.lang.annotation.AfterReturning;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Before;
+import org.aspectj.lang.annotation.Pointcut;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.boot.autoconfigure.freemarker.FreeMarkerTemplateAvailabilityProvider;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+import static com.library.constant.ApplicationConstant.DATE_TIME_FORMATTER;
+
+@Aspect
+@Component
+@RequiredArgsConstructor
+public class RabbitProducer {
+
+    private final RabbitProperties rabbitProperties;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RabbitProducer.class);
+
+    private final RabbitTemplate rabbitTemplate;
+
+    @Pointcut("execution(public * com.library.controller.BookController.*(..))")
+    public void callAtBookControllerPublic() { }
+
+    @Pointcut("execution(public * com.library.controller.DefaultExceptionHandler.*(..))")
+    public void callAtExceptionHandlerPublic() { }
+
+
+    @Before("callAtBookControllerPublic()")
+    public void sendRequiredArguments(JoinPoint joinPoint){
+        String s = Arrays.stream(joinPoint.getArgs()).map(Object::toString).collect(Collectors.joining(", "));
+        String message = String.format("%s - %s - args: %s",LocalDateTime.now().format(DATE_TIME_FORMATTER), joinPoint.getStaticPart().getSignature(), s);
```

Comment:
Dont see the names of args. How it works and how result message is looked?

Reply:
Names of args are invisible, here I mark message what type it belongs to